pkg/splunk/enterprise/configuration.go

```diff
@@ -929,17 +929,15 @@ func validateSplunkAppSources(appFramework *enterpriseApi.AppFrameworkSpec, loca
 		}
 
 		if appSrc.Scope != "" {
-			if localScope && appSrc.Scope != "local" {
+			if localScope && appSrc.Scope != enterpriseApi.ScopeLocal {
 				return fmt.Errorf("Invalid scope for App Source: %s. Only local scope is supported for this kind of CR", appSrc.Name)
 			}
 
-			if appSrc.Scope != "local" && appSrc.Scope != "cluster" {
+			if !(appSrc.Scope == enterpriseApi.ScopeLocal || appSrc.Scope == enterpriseApi.ScopeCluster || appSrc.Scope == enterpriseApi.ScopeClusterWithPreConfig) {
 				return fmt.Errorf("Scope for App Source: %s should be either local or cluster", appSrc.Name)
```

Comment:
Probably modify the error msg too, something like - "Scope for App Source: %s should be either local or cluster or clusterWithPreConfig"